lib/asm-parser.js

```diff
@@ -318,10 +355,10 @@ class AsmParser extends AsmRegex {
             }
 
             if (filters.libraryCode && !lastOwnSource && source && source.file !== null) {
-                if (mayRemovePreviousLabel && result.length > 0) {
-                    const lastLine = result[result.length - 1];
+                if (mayRemovePreviousLabel && astResult.length > 0) {
+                    const lastLine = astResult[astResult.length - 1];
                     if (lastLine.text && lastLine.text.match(this.labelDef)) {
-                        result.pop();
+                        astResult.pop();
```

Comment:
I also did some refactoring here. If the above condition is true, we will remove the last line from the ast result which contains a label definition. In this case we can remove the label definition from the `labelDefinitions` map too.

I updated the test cases.